jetty-ee10/jetty-ee10-servlet/src/test/java/org/eclipse/jetty/ee10/servlet/DefaultServletCombinationsTest.java

```diff
@@ -0,0 +1,191 @@
+            new Data(true, SERVE, "/", HttpStatus.FOUND_302, "http://local/index.html/"),
+            new Data(true, REDIRECT, "/", HttpStatus.FOUND_302, "http://local/index.html"),
+            new Data(true, REHANDLE, "/", HttpStatus.FOUND_302, "http://local/index.html/"),
+
+            new Data(false, SERVE, "/index.html", HttpStatus.OK_200, "Static index.html at root"),
+            new Data(false, REDIRECT, "/index.html", HttpStatus.OK_200, "Static index.html at root"),
+            new Data(false, REHANDLE, "/index.html", HttpStatus.OK_200, "Static index.html at root"),
+            new Data(true, SERVE, "/index.html", HttpStatus.FOUND_302, "http://local/index.html/"),
+            new Data(true, REDIRECT, "/index.html", HttpStatus.FOUND_302, "http://local/index.html/"),
+            new Data(true, REHANDLE, "/index.html", HttpStatus.FOUND_302, "http://local/index.html/"),
+
+            new Data(false, SERVE, "/foo.welcome", HttpStatus.OK_200, "Servlet at welcome extension"),
+            new Data(false, REDIRECT, "/foo.welcome", HttpStatus.OK_200, "Servlet at welcome extension"),
+            new Data(false, REHANDLE, "/foo.welcome", HttpStatus.OK_200, "Servlet at welcome extension"),
+            new Data(true, SERVE, "/foo.welcome", HttpStatus.OK_200, "Servlet at welcome extension"),
+            new Data(true, REDIRECT, "/foo.welcome", HttpStatus.OK_200, "Servlet at welcome extension"),
+            new Data(true, REHANDLE, "/foo.welcome", HttpStatus.OK_200, "Servlet at welcome extension"),
+
+            new Data(false, SERVE, "/subdir/", HttpStatus.OK_200, "Static index.html at subdir"),
+            new Data(false, REDIRECT, "/subdir/", HttpStatus.FOUND_302, "http://local/subdir/index.html"),
+            new Data(false, REHANDLE, "/subdir/", HttpStatus.OK_200, "Static index.html at subdir"),
+            new Data(true, SERVE, "/subdir/", HttpStatus.FOUND_302, "http://local/subdir/index.html/"),
+            new Data(true, REDIRECT, "/subdir/", HttpStatus.FOUND_302, "http://local/subdir/subdir/index.html"),
+            new Data(true, REHANDLE, "/subdir/", HttpStatus.FOUND_302, "http://local/subdir/index.html/"),
+
+            new Data(false, SERVE, "/subdir/foo.welcome", HttpStatus.OK_200, "Servlet at welcome extension"),
+            new Data(false, REDIRECT, "/subdir/foo.welcome", HttpStatus.OK_200, "Servlet at welcome extension"),
+            new Data(false, REHANDLE, "/subdir/foo.welcome", HttpStatus.OK_200, "Servlet at welcome extension"),
+            new Data(true, SERVE, "/subdir/foo.welcome", HttpStatus.OK_200, "Servlet at welcome extension"),
+            new Data(true, REDIRECT, "/subdir/foo.welcome", HttpStatus.OK_200, "Servlet at welcome extension"),
+            new Data(true, REHANDLE, "/subdir/foo.welcome", HttpStatus.OK_200, "Servlet at welcome extension"),
+
+            new Data(false, SERVE, "/nothing/index.welcome", HttpStatus.OK_200, "Servlet at welcome extension"),
```

Comment:
as well as "/nothing/", which is a directory that does not exist, we need "/empty/" which is what it says it is!